controllers/awsclusterstatic_identity_controller.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/scope"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/cluster-api/util/predicates"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// AWSClusterStaticIdentityReconciler reconciles a AWSClusterIdentity object.
+type AWSClusterStaticIdentityReconciler struct {
+	client.Client
+	Log              logr.Logger
+	Endpoints        []scope.ServiceEndpoint
+	WatchFilterValue string
+}
+
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsclusterstaticidentities,verbs=get;list;watch;update
+
+func (r *AWSClusterStaticIdentityReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Fetch the AWSCluster instance
+	awsCluster := &infrav1.AWSCluster{}
+	err := r.Get(ctx, req.NamespacedName, awsCluster)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Fetch all static identities
+	staticIdentities := &infrav1.AWSClusterStaticIdentityList{}
+	err = r.List(ctx, staticIdentities)
```

Comment:
After any changes (and periodically), `AWSClusterStaticIdentity` object will start reconciling. We don't need to list all identities here.
Similar to other controllers, get the object from the request and only do the checks on that particular resource.

Reply:
The problem with that approach is we will be able to get the identity object, but we wont be able to get the cluster object from request, as identities are cluster scoped, there wont be any namespaces

Reply:
Right, I remember now. Cluster-scoped objects should not have namespace scoped owner references normally. 

I wonder if this becomes a problem in terms of Garbage collection.
Since cluster-scoped identity will have owner refs that is not cluster-scoped, garbage collector will not be able to find the owners and tries to delete the object? Not sure, but since you did tests I am guessing something is blocking this to happen, probably finalizer. 
WDYT is happening?

Reply:
This is a relatively new change, which makes the identities with absent owners not to be garbage collected:

```
Cluster-scoped dependents can only specify cluster-scoped owners. In v1.20+, if a cluster-scoped dependent specifies a namespaced kind as an owner, it is treated as having an unresolvable owner reference, and is not able to be garbage collected.
```

Reply:
After brainstorming a bit, I found that we can use only annotations or addition of new field to the identity object (like list of owners) to make sure deletion goes through properly.
In any case, we can't use ownerReferences if the owner and dependent are not in same scope.